Don't use gcc specific flags, because it causes various undefined
reference errors with ports-gcc>=6 when -lsupc++ is used.

Add --enable-24-bit-color, --enable-wide-glyphs and --enable-harfbuzz.

Index: configure.ac
--- configure.ac.orig
+++ configure.ac
@@ -28,25 +28,6 @@
 AC_LANG(C++)
 
 LINKER="$CXX"
-dnl check whether we can link with gcc -lsupc++
-if test x$GCC = xyes && test x$GXX = xyes; then
-   dnl FreeBSD (at least up to 5.3) has a broken GCC, workaround it
-   AC_MSG_CHECKING([for working libsupc++])
-   save_CXX="$CXX"
-   save_LIBS="$LIBS"
-   CXX="$CC"
-   LIBS="$LIBS -lsupc++"
-   AC_LINK_IFELSE(
-      [AC_LANG_PROGRAM([struct test { }; void f() try { throw new test; } catch (...) { throw; }],[])],
-      [AC_MSG_RESULT(ok)
-       LINKER="$CC"],
-      [
-         AC_MSG_RESULT([no, making everything bigger and slower])
-         LIBS="$save_LIBS"
-      ]
-   )
-   CXX="$save_CXX"
-fi
 AC_SUBST(LINKER,[$LINKER])
 
 dnl# --------------------------------------------------------------------------
@@ -90,6 +71,7 @@
 support_8bitctrls=no
 support_iso14755=yes
 support_styles=yes
+support_harfbuzz=no
 support_perl=yes
 codesets=all
 
@@ -123,6 +105,8 @@
        support_8bitctrls=no
        support_iso14755=no
        support_styles=no
+       support_wide_glyphs=no
+       support_harfbuzz=no
        support_perl=no
        codesets=
     fi
@@ -148,6 +132,8 @@
        #support_8bitctrls=yes
        support_iso14755=yes
        support_styles=yes
+       support_wide_glyphs=yes
+       support_harfbuzz=yes
        support_perl=yes
        codesets=all
     fi
@@ -197,6 +183,16 @@
   AC_DEFINE(USE_256_COLORS, 1, Define if you want 256-color support)
 fi
 
+support_24_bit_color=no
+AC_ARG_ENABLE(24-bit-color,
+  [  --enable-24-bit-color   enable 24-bit color support],
+  [if test x$enableval = xyes; then
+    support_24_bit_color=yes
+  fi])
+if test x$support_24_bit_color = xyes; then
+  AC_DEFINE(USE_24_BIT_COLOR, 1, Define if you want 24-bit color support)
+fi
+
 AC_ARG_ENABLE(unicode3,
   [  --enable-unicode3       use 21 instead of 16 bits to represent unicode characters],
   [if test x$enableval = xyes -o x$enableval = xno; then
@@ -221,6 +217,18 @@
     support_styles=$enableval
   fi])
 
+AC_ARG_ENABLE(wide-glyphs,
+  [  --enable-wide-glyphs    enable displaying of wide glyphs],
+  [if test x$enableval = xyes -o x$enableval = xno; then
+    support_wide_glyphs=$enableval
+  fi])
+
+AC_ARG_ENABLE(harfbuzz,
+  [  --enable-harfbuzz       enable text shaping (ligatures) with harfbuzz],
+  [if test x$enableval = xyes -o x$enableval = xno; then
+    support_harfbuzz=$enableval
+  fi])
+
 AC_ARG_ENABLE(pixbuf,
   [  --enable-pixbuf         enable integration with gdk-pixbuf for background images],
   [if test x$enableval = xyes -o x$enableval = xno; then
@@ -619,6 +627,14 @@
   fi
 fi
 
+if test x$support_xft = xyes && test x$support_harfbuzz = xyes; then
+  RXVT_CHECK_MODULES([HARFBUZZ], [harfbuzz], [
+    X_LIBS="$HARFBUZZ_LIBS $X_LIBS"
+    CPPFLAGS="$CPPFLAGS $HARFBUZZ_CFLAGS"
+    AC_DEFINE(HAVE_HARFBUZZ, 1, Define to enable text shaping with harfbuzz)
+  ], [support_harfbuzz=no])
+fi
+
 if test x$support_image = xyes && test x$rxvt_have_xrender = xyes; then
   AC_MSG_CHECKING(for Render >= 0.11)
   AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[
@@ -643,6 +659,9 @@
 if test x$support_styles = xyes; then
   AC_DEFINE(ENABLE_STYLES, 1, Define if you want bold and italic support)
 fi
+if test x$support_wide_glyphs = xyes; then
+  AC_DEFINE(ENABLE_WIDE_GLYPHS, 1, Define if you want to display wide glyphs)
+fi
 if test x$support_iso14755 = xyes; then
   AC_DEFINE(ISO_14755, 1, Define if you want ISO 14755 extended support)
 fi
